doc/ch-image_desc.rst

```diff
@@ -88,6 +91,34 @@ Common options placed before the sub-command:
   :code:`-v`, :code:`--verbose`
     Print extra chatter; can be repeated.
 
+Authentication
+==============
+
+If the remote repository needs authentication, Charliecloud will prompt you
+for a username and password. Note that some repositories call the secret
+something other than "password"; e.g., GitLab calls it a "personal access
+token (PAT)".
+
+These values are remembered for the life of the process and silently
+re-offered to the registry if needed. One case when this happens is on push to
+a private registry: many registries will first offer a read-only token when
+:code:`ch-image` checks if something exists, then re-authenticate when
+upgrading the token to read-write for upload. If your site uses one-time
+passwords such as provided by a security device, you can specify
+:code:`--password-once` to provide a new secret each time.
+
+These values are not saved persistently, e.g. in a file. Note that we do use
+normal Python variables for this information, without pinning them into
+physical RAM with `mlock(2)
+<https://man7.org/linux/man-pages/man2/mlock.2.html>`_ or any other special
+treatment, so we cannot guarantee they will never reach non-volatile storage.
```

Comment:
@mej Do you have any thoughts on how risky this is?

Reply:
For a little more context, it seemed impossible to reliably `mlock(2)` Python variables. The closest it seemed I could come was to allocate a raw buffer and put the data in there, but then I couldn't get it in or out without going through Python variables anyway.